src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Protocol.cs

```diff
@@ -30,30 +30,18 @@ internal sealed class SecureChannel
         private int _trailerSize = 16;
         private int _maxDataSize = 16354;
 
-        private bool _refreshCredentialNeeded;
-
-        private readonly SslAuthenticationOptions _sslAuthenticationOptions;
-        private SslApplicationProtocol _negotiatedApplicationProtocol;
+        private bool _refreshCredentialNeeded = true;
 
         private static readonly Oid s_serverAuthOid = new Oid("1.3.6.1.5.5.7.3.1", "1.3.6.1.5.5.7.3.1");
         private static readonly Oid s_clientAuthOid = new Oid("1.3.6.1.5.5.7.3.2", "1.3.6.1.5.5.7.3.2");
-        private SslStream? _ssl;
 
-        internal SecureChannel(SslAuthenticationOptions sslAuthenticationOptions, SslStream sslStream)
+        static SslStream()
```

Comment:
Can we avoid this?  On non-Windows this is going to be pure overhead.

Reply:
I was thinking about it. Now we call the validate for every session but it seems static so I would expect the result would be always same. And I'm not sure if we really needed. I'm wondering if this is old code from time when TLS was optional.

Reply:
I did more testing and everything seems to work fine so I decided to remove it.